### PR TITLE
#29 feat: added password input field in user module

### PR DESCRIPTION
**Pull request by vallo87**
password input field in users module

### CODE REVIEW

File: config/.env.default [hunk 1]
```diff
@@ -17,8 +17,8 @@ export APP_DEFAULT_LOCALE="en_US"
 export SECURITY_SALT="__SALT__"
 
 # Uncomment these to define BEDITA API base URL and API KEY
-export BEDITA_API="https://bedita-api-url"
-export BEDITA_API_KEY="bedita-api-key"
+# export BEDITA_API="https://bedita-api-url"
```
Comment (stefanorosanelli) [inline config/.env.default]: This `.env.default` file will be copied in `.env` and those env vars are mandatory, as explained in README.md => I would leave them uncommented (they are not optional).

File: src/Template/Element/Form/other_properties.twig [hunk 2]
```diff
@@ -1,26 +1,110 @@
...
-$(document).ready(function(){
-    $('textarea.json').change(function(event){
-        if (event && event.target && event.target.value) {
-            try {
-                let jsonString = JSON.stringify(JSON.parse(event.target.value), null, "    ");
-                $(event.target).val(jsonString);
-            } catch(e) {
-                console.error(e);
-            }
+
```
Comment (stefanorosanelli) [inline src/Template/Element/Form/other_properties.twig]: What about a separate `Element/Form/_other_properties_script.twig` file containing JS code?
Here jus `{% element 'Form/_other_properties_script' %}`

File: src/Template/Element/Form/other_properties.twig [hunk 3]
```diff
@@ -1,26 +1,110 @@
...
+        confirm_password.closest('div').removeAttribute('message-tooltip');
+    }
+
+    let validateForm = (event, callback) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        let form = document.getElementById('form--main');
+        let passwordGroup = form.elements.password;
+        let password = passwordGroup[0];
```
Comment (stefanorosanelli) [inline src/Template/Element/Form/other_properties.twig]: This causes a JS error on `Save` in other modules like `Documents` or any other where no `password` field is present.

```
14085:380 Uncaught TypeError: Cannot read property '0' of undefined
    at HTMLInputElement.validateForm (14085:380)
validateForm @ 14085:380
```

Comment (stefanorosanelli) [inline src/Template/Element/Form/other_properties.twig]: This causes a JS error on `Save` in other modules like `Documents` or any other where no `password` field is present.

```
14085:380 Uncaught TypeError: Cannot read property '0' of undefined
    at HTMLInputElement.validateForm (14085:380)
validateForm @ 14085:380
```

File: src/Template/Element/Form/other_properties.twig [hunk 4]
```diff
@@ -37,6 +121,27 @@ $(document).ready(function(){
...
+            {% if key == 'username' and currentModule.name == 'users' %}
+                <div class="password-component">
+                    {{ Form.control('password', {
+                        'class': 'password',
+                        'placeholder': 'new password',
+                        'autocomplete': 'new-password',
+                        'default': ''
+                    })|raw }}
+
+                    {{ Form.control('password', {
```
Comment (stefanorosanelli) [inline src/Template/Element/Form/other_properties.twig]: This causes a DOM error:
```
[DOM] Found 2 elements with non-unique id #password: (More info: https://goo.gl/9p2vKq) <input type=​"password" name=​"password" class=​"password" placeholder=​"new password" autocomplete=​"new-password" id=​"password" value>​ <input type=​"password" name=​"confirm_password" class=​"confirm-password" placeholder=​"confirm password" autocomplete=​"new-password" id=​"password" value>​
```

File: src/Template/Element/Form/other_properties.twig [hunk 5]
```diff
@@ -1,26 +1,110 @@
...
-                let jsonString = JSON.stringify(JSON.parse(this.value), null, "    ");
-                $(this).val(jsonString);
-            } catch(e) {
-                console.error(e);
+        return true;
+    }
+
+    let cleanupForm = () => {
+        // reset messages/styles for password validation
+        let form = document.getElementById('form--main');
```
Comment (didoda) [inline src/Template/Element/Form/other_properties.twig]: con jquery potresti scrivere: `let form = $('#form--main')[0];`

File: src/Template/Element/Form/other_properties.twig [hunk 6]
```diff
@@ -1,26 +1,110 @@
...
+        let confirm_password = form.elements.confirm_password;
+        confirm_password.classList.remove('error', 'empty', 'different');
+        confirm_password.closest('div').removeAttribute('message-tooltip');
+    }
+
+    let validateForm = (event, callback) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        let form = document.getElementById('form--main');
```
Comment (didoda) [inline src/Template/Element/Form/other_properties.twig]: come sopra, `let form = $('#form--main')[0];`

File: src/Template/Element/Form/other_properties.twig [hunk 7]
```diff
@@ -1,26 +1,110 @@
...
+
+        form.submit();
+    }
+
+    $(document).ready(function(){
+
+        // page events
+
+        // validate form on submit
+        let saveButton = document.querySelector('input[type="submit"]');
```
Comment (didoda) [inline src/Template/Element/Form/other_properties.twig]: or `let saveButton = $('input[type="submit"]);`

**Pull request merged**